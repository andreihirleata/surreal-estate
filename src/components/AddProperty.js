import React from "react";

const AddProperty = () => {
  return (
    <div>
      <p>Add Property Page</p>
    </div>
  );
};

export default AddProperty;
