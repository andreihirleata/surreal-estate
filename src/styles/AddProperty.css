.add-property {
  margin: auto;
  padding: 15px 15px;
  flex-direction: column;
  align-items: center;
  min-height: fit-content;
  width: 400px;
  background-color: white;
  -webkit-box-shadow: 6px -4px 5px 1px rgba(0, 0, 0, 0.66);
  -moz-box-shadow: 6px -4px 5px 1px rgba(0, 0, 0, 0.66);
  box-shadow: 6px -4px 5px 1px rgba(0, 0, 0, 0.66);
}

.wrapper {
  height: 100vh;
  background-color: #f0f0f0;
  padding-top: 50px;
}
.form-entry {
  margin: 15px;
  text-align: center;
  font-weight: 700;
  display: flex;
  flex-direction: column;
  align-content: center;
}

label {
  margin-right: 10px;
  margin: auto;
}

button {
  margin: auto;
  width: 100px;
  height: 30px;
  /* border-radius: 25px; */
  text-decoration: none;
  border: none;
  font-weight: 700;
  box-shadow: none;
  background-color: purple;
  color: white;
  transition: background 250ms ease-in-out, transform 150ms ease;
}
button:active {
  transform: scale(0.99);
}

input {
  margin: auto;
  width: 70%;
  height: 30px;
  text-align: center;
}

select {
  width: 70%;
  height: 30px;
  text-align-last: center;
  margin: auto;
}
