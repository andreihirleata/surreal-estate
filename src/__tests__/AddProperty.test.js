import React from "react";
import { render } from "@testing-library/react";
import AddProperty from "../components/AddProperty";

describe("AddProperty", () => {
  test("page matches snapshot", () => {
    const { asFragment } = render(<AddProperty />);
    const component = asFragment();

    expect(component).toMatchSnapshot();
  });

  test("renders AddProperty component ", () => {
    const { getByTestId, getByLabelText, getByRole } = render(<AddProperty />);

    const mainDiv = getByTestId("add-property");

    expect(mainDiv).toBeInTheDocument();
    expect(getByRole("button")).toHaveTextContent(/add/i);
    expect(getByLabelText(/title/i)).toBeInTheDocument();
    expect(getByLabelText(/city/i)).toBeInTheDocument();
    expect(getByLabelText(/type/i)).toBeInTheDocument();
    expect(getByLabelText(/bedrooms/i)).toBeInTheDocument();
    expect(getByLabelText(/bathrooms/i)).toBeInTheDocument();
    expect(getByLabelText(/price/i)).toBeInTheDocument();
    expect(getByLabelText(/email/i)).toBeInTheDocument();
  });
});
