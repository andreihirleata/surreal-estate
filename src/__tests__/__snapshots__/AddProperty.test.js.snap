// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddProperty page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      class="add-property"
      data-testid="add-property"
    >
      <form>
        <div
          class="form-entry"
        >
          <label
            for="title"
          >
            Title:
          </label>
          <input
            id="title"
            name="title"
            value=""
          />
        </div>
        <div
          class="form-entry"
        >
          <label
            for="type"
          >
            Type:
          </label>
          <select
            id="type"
            name="type"
          >
            <option
              value="Flat"
            >
              Flat
            </option>
            <option
              value="Detached"
            >
              Detached
            </option>
            <option
              value="Semi-Detached"
            >
              Semi-Detached
            </option>
            <option
              value="Terraced"
            >
              Terraced
            </option>
            <option
              value="End of Terrace"
            >
              End of Terrace
            </option>
            <option
              value="Cottage"
            >
              Cottage
            </option>
            <option
              value="Bungalow"
            >
              Bungalow
            </option>
          </select>
        </div>
        <div
          class="form-entry"
        >
          <label
            for="bedrooms"
          >
            Bedrooms:
          </label>
          <input
            id="bedrooms"
            name="bedrooms"
            value=""
          />
        </div>
        <div
          class="form-entry"
        >
          <label
            for="bathrooms"
          >
            Bathrooms:
          </label>
          <input
            id="bathrooms"
            name="bathrooms"
            value=""
          />
        </div>
        <div
          class="form-entry"
        >
          <label
            for="price"
          >
            Price:
          </label>
          <input
            id="price"
            name="price"
            value=""
          />
        </div>
        <div
          class="form-entry"
        >
          <label
            for="city"
          >
            City:
          </label>
          <select
            id="city"
            name="city"
          >
            <option
              value="Manchester"
            >
              Manchester
            </option>
            <option
              value="Leeds"
            >
              Leeds
            </option>
            <option
              value="Sheffield"
            >
              Sheffield
            </option>
            <option
              value="Liverpool"
            >
              Liverpool
            </option>
          </select>
        </div>
        <div
          class="form-entry"
        >
          <label
            for="email"
          >
            Email:
          </label>
          <input
            id="email"
            name="email"
            value=""
          />
        </div>
        <div
          class="form-entry"
        >
          <button
            type="submit"
          >
            Add
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
